returnn/tf/layers/basic.py

```diff
@@ -2986,7 +2995,9 @@ def get_out_data_from_opts(cls, name, base, sources=(), **kwargs):
     data = get_concat_sources_data_template(sources).copy_as_batch_major()
     data.name = "%s_output" % name
     assert data.batch_dim_axis == 0
+    data.batch = base.output.batch
     data.time_dim_axis = 1
+    data.size_placeholder = base.output.size_placeholder.copy()
```

Comment:
Yea. But this can be a separate change, to make `base` optional

But as you comment on this line, I just noticed, I was a bit sloppy here. I should have moved the logic from `__init__` here:
```
    seq_lens = base.output.get_sequence_lengths()
    data.size_placeholder[0] = seq_lens
```
And also cleanup that `size_placeholder` logic in `__init__`.